### ISSUE TITLE
Enhancement: [prefer-nullish-coalescing] Add runtime check for strictNullChecks being enabled

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5995]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-nullish-coalescing

### Description

Following up from https://github.com/typescript-eslint/typescript-eslint/issues/5748#issuecomment-1270859149:

> We should give [prefer-nullish-coalescing] the same treatment as rules like strict-boolean-expressions where the rule will explicitly error if you don't have the option turned on.
>
> https://typescript-eslint.io/rules/strict-boolean-expressions/#allowruletorunwithoutstrictnullchecksiknowwhatiamdoing
>
> I would say we can do that change without a breaking change because the rule is LITERALLY useless without the option.

### Fail

```typescript
n/a
```

### Pass

```typescript
n/a
```

### Additional Info

#5988 added docs; now let's add a runtime check. 🚀

**Comment by JoshuaKGoldberg** [comment]
@cparros mentioned interest in tackling this soon. 😁

**Comment by JoshuaKGoldberg** [comment]
Fixed by #6174.
